backend/src/api/export.rs

```diff
@@ -1,13 +1,15 @@
 use crate::models::{Metadata, MetadataReadback, Well};
 use async_graphql::Object;
 use itertools::Itertools;
+use opa_client::graphql::OPAGuard;
 use soakdb::{insert_wells, write_metadata};
 
 #[derive(Debug, Default)]
 pub struct ExportMutation;
 
 #[Object]
 impl ExportMutation {
+    #[graphql(guard = "OPAGuard::new(\"xchemlab.soakdb_interface.allow\")")]
```

Comment:
Are you comfortable baking in the use of `OPAGuard` here? Could hide it behind a trait and leave the option to add a built-in auth mechanism as an alternative. Only matters if this service should be able to operate fully standalone.

Reply:
Pretty happy to bake it in, I expect other resolvers will have it even more baked-in (e.g. if they require resource lookup before an AuthZ decision can be made)

Reply:
Though would also be open to abstract authz if you have an idea for a simple concrete implementation. I'm struggling to think how I'd make it work

Reply:
Could just do something that looks a lot like OPA as a first pass, but probably not worth the time at this stage